### PR TITLE
Wrap FOOOF

### PR DESCRIPTION
**Pull request by palday**
One big issue that I found is that FOOOF is very sensitive to being passed Python lists vs NumPy arrays. PyCall is very aggressive about converting to NumPy arrays, so this creates problems. Given Python's duck-typing philosophy, this seems like something that might make sense open PRs for in upstream, but I'm not sure.  

TODO:
- [x] README (use PyMNE as a starting point)
   - [x] general usage
   - [x] forcing arguments to be a list or a list of lists
   - [x] badges
      - [x] codecov
      - [x] ci
- [x] remove Onda-related functionality 
- [x] examine whether it makes sense to keep the `AbstractArray` functionality

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+uuid = "bb993ee8-2a51-46a1-a315-55ff424041b4"
+authors = ["Beacon Biosignals, Inc."]
+version = "0.1.0"
+
+[deps]
+PyCall = "438e738f-606a-5dbb-bf0a-cddfbfd45ab0"
+PyPlot = "d330b81b-6aea-500a-939a-2ce795aea3ee"
+
+[compat]
+PyCall = "1.90"
```
Comment (ararslan) [inline Project.toml:11]: Missing a `[compat]` entry for PyPlot

File: README.md [hunk 2]
```diff
@@ -1,2 +1,249 @@
...
+by PyCall. If you're using PyCall via its hidden Miniconda install, your own
+Anaconda environment, or a Python virtual environment, this is what you want.
+(The "global" path is sandboxed to the Conda/virtual environment.) If you are
+however using system Python, then you should set `ENV["PIPFLAGS"] = "--user"`
+before `add`ing / `build`ing the package. By default, PyFOOOF will use the latest
+FOOOF 1.x release available on [PyPI](https://pypi.org/project/FOOOF/), but this can also
+be changed via the `ENV["FOOOFVERSION"] = version_number` for your preferred
+`version_number`.
+
+Note that FOOOF-Python uses [`matplotlib`](https://matplotlib.org/) for plotting, but does not install it automatically as a dependency.
```
Comment (ararslan) [inline README.md:28]: Doesn't PyPlot install matplotlib?

File: README.md [hunk 3]
```diff
@@ -1,2 +1,249 @@
...
+If you wish to take advantage of this functionality, the non-exported `install_matplotlib` function will install `matplotlib`, using the same environment variables as the main installation.
+The Julia package [`PyPlot`](https://github.com/JuliaPy/PyPlot.jl), which provides a Julia interface to `matplotlib`, is also useful for manipulating the rendered plots.
+
+FOOOF-Python can also be installed manually ahead of time.
+From the shell, use `python -m pip install fooof` for the latest stable release
+or `python -m pip install fooof==version_number` for a given `version_number`,
+ensuring  that `python` is the same one that PyCall is using. Alternatively,
+you can run this from within Julia:
+```julia
+julia> using PyCall
```
Comment (ararslan) [inline README.md:38]: Inconsequential nitpick, but the fake REPL sessions in docs bother me (i.e. ones that don't show output so you know they're editied or falsified). I'd just remove the prompts.

File: src/PyFOOOF.jl [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+#####
+
+const fooof = PyNULL()
+
+function __init__()
+    # all of this is __init__() so that it plays nice with precompilation
+    # see https://github.com/JuliaPy/PyCall.jl/#using-pycall-from-julia-modules
+    copy!(fooof, pyimport("fooof"))
+    # don't eval into the module while precompiling; this breaks precompilation
+    # of downstream modules (see #4)
```
Comment (ararslan) [inline src/PyFOOOF.jl:20]: The issue reference here is for PyMNE, not this repo
```suggestion
    # of downstream modules
```

File: README.md (hunk 3 above)
Comment (palday) [reply README.md:38]: The prompts are here because there are also Python examples

File: README.md (hunk 2 above)
Comment (palday) [reply README.md:28]: This gets complicated .... but actually we don't need the PyPlot dependency for the core functionality, so I'll drop it and leave it up to the user how they want to handle it

File: README.md [hunk 5]
```diff
@@ -1,2 +1,257 @@
...
+
+However, naively using a true 2d array (`Matrix`) in Julia also results in error:
+```julia
+julia> gauss_params = [10 1.0 2.5; 20 0.8 2; 32 0.6 1]
+3×3 Matrix{Float64}:
+ 10.0  1.0  2.5
+ 20.0  0.8  2.0
+ 32.0  0.6  1.0
+julia> freqs, spectrum = gen_power_spectrum(f_range, ap_params, gauss_params, nlv)
+\ERROR: PyError ($(Expr(:escape, :(ccall(#= /home/ubuntu/.julia/packages/PyCall/BD546/src/pyfncall.jl:43 =# @pysym(:PyObject_Call), PyPtr, (PyPtr, PyPtr, PyPtr), o, pyargsptr, kw))))) <class 'ValueError'>
```
Comment (ararslan) [inline README.md:238]: Does the error message here actually start with a backslash? And is the line below missing a closing )?

File: src/PyFOOOF.jl [hunk 6]
```diff
@@ -0,0 +1,52 @@
+module PyFOOOF
+
+using PyCall
+
+#####
+##### Exports
+#####
+
```
Comment (ararslan) [inline src/PyFOOOF.jl:8]:
```suggestion
```
Might as well delete this section

File: src/PyFOOOF.jl [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+            isdefined(@__MODULE__, pn) && continue
+            prop = getproperty(fooof, pn)
+            @eval $pn = $prop
+        end
+    end
+    return nothing
+end
+
+"""
+    install_matplotlib(ver="")
```
Comment (ararslan) [inline src/PyFOOOF.jl:33]: I feel like installing matplotlib should just be the user's responsibility. That is, it seems odd to me for a package meant to wrap FOOOF to offer to install a different package. We could direct users to PyPlot if they want to interface with FOOOF's matplotlib functionality.

**Review by ararslan — APPROVED** [review]

**Pull request merged**